src/transaction.ts

```diff
@@ -1073,18 +1075,53 @@ export function decodeUnsignedTransaction(
   return Transaction.from_obj_for_encoding(partlyDecodedObject);
 }
 
+/**
+ * Object representing a transaction with a signature
+ */
+export interface SignedTransaction {
```

Comment:
This should be moved to the `types` directory. Perhaps `src/types/transactions/base.ts` would make sense?

Reply:
Unfortunately placing this type elsewhere would create a dependency cycle, since it needs to import `Transaction`. We could move `decodeSignedTransaction` to a different file to break this loop though, what do you think about that?

Reply:
That's a tough one. I'm hesitant to relocate only that function to a new file, but it feels bad to store a widely-used type somewhere other than the `types` directory. Maybe the solution is to leave it here for now and relocate it later once we have a "transaction auxiliary functions" file when a sufficient number of functions similar to `decodeSignedTransaction` have been added.